packages/flutter/lib/src/semantics/semantics.dart

```diff
@@ -3958,6 +3951,14 @@ class SemanticsNode with DiagnosticableTreeMixin {
     final Float64List updatedTransform;
     if (kIsWeb) {
       updatedTransform = data.transform?.storage ?? _kIdentityTransform;
+    } else if (_isTraversalChild) {
+      updatedTransform = _computeChildTransform(
```

Comment:
We can move line 3940 to here as well because `traversalParent` is only for the traversal child transform recalculation. And then we can remove that if statement.